relay-kafka/src/producer/mod.rs

```diff
@@ -80,16 +80,29 @@ pub trait Message {
 }
 
 /// Single kafka producer config with assigned topic.
-struct SingleProducer {
+struct Producer {
+    /// Time of the latest collection of stats.
+    last_report: Cell<Instant>,
     /// Kafka topic name.
     topic_name: String,
     /// Real kafka producer.
     producer: Arc<ThreadedProducer>,
 }
 
-impl fmt::Debug for SingleProducer {
+impl Producer {
+    fn new(topic_name: String, producer: Arc<ThreadedProducer>) -> Self {
+        Self {
+            last_report: Instant::now().into(),
```

Comment:
```suggestion
            last_report: Cell::new(Instant::now()),
```

Personal opinion, I'd use `into` only if the type isn't in scope.

Reply:
Agree, explicit is better than implicit.